tutorial.md

```diff
@@ -0,0 +1,1022 @@
+    someone who actually uses a debian-derivative has to confirm them.
+
+  * Switch all of the links to contrib 0.17 when it's out.
+
+  * Someone who uses trayer needs to look over this because I have no
+    idea and simply copied most of the old tutorial there.
+
+  * The closing thoughts could probably be a bit more extensive.
+
+QUESTIONS:
+
+  * Lots of code is very similar here, only sporting small adjustments.
+    It might be more natural working with `GNU diff` files here, instead
+    of copy-pasting slightly changed code over and over again.  This may
+    or may not confuse beginners, needs feedback.
+
+  * Is anything missing that beginners really ought to know about?
+
+  * Is this comprehensible for people who don't know Haskell?
+
+  * I wonder if the xmobar config should use font-awesome or something
+    similar or if that's overdoing it.  People really like eye-candy.
+    On that note, the xmobar configuration might be a little too much
+    (the xmonad `PP` is basically a copy-paste of my personal settings),
+    but perhaps this will give people the start they need and then they
+    can focus on more important things than how their bar looks :)
+
+  * Is `xscreensaver` still relevant?
+
+
+# XMonad Configuration Tutorial
+
+I'm going to take you, step-by-step, through the process of configuring
+xmonad, setting up a status bar with [xmobar], setting up a tray with
+[trayer-srg], and making it all play nicely together.
+
+I assume that you have read the [xmonad guided tour] already.  It is a
+bit dated at this point but, because xmonad is stable, the guide should
+still give you a good overview of the most basic functionality.
+
+Before we begin, here are two screenshots of the kind of desktop we will
+be creating together.  In particular, a useful layout we'll conjure into
+being—the three columns layout from [XMonad.Layout.ThreeColumns] with
+the ability to magnify stack windows with [XMonad.Layout.Magnifier]:
+
+<p>
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111586872-c0eb0e00-87c1-11eb-9129-c2781bbbf227.png" width="550">
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111586885-c47e9500-87c1-11eb-8e8e-83d4d33c4c8d.png" width="550">
+</p>
+
+So let's get started!
+
+## Preliminaries
+
+First you'll want to install xmonad.  You can either do this with your
+systems package manager, or (a bit more advanced) via `stack`/`cabal`.
+You can find instructions for the latter in the [xmonad-testing]
+repository.  I'm going to assume xmonad version `0.16` and
+xmonad-contrib version `0.17` here, though most of these steps should
+work with older versions as well.  When we get to the relevant parts,
+will point you to alternatives that work with at least xmonad version
+`0.15` and xmonad-contrib version `0.16`.  This will usually be
+accompanied by a big fat "_IF YOU ARE ON A VERSION `< 0.17`_", so don't
```

Comment:
```suggestion
accompanied by a big "_IF YOU ARE ON A VERSION `< 0.17`_", so don't
```

Language? I don't think it's something I'd be at ease reading in the xmonad manual.

Reply:
That's fair enough; I probably got a bit too carried away there :)